test/stdlib/KeyPath.swift

```diff
@@ -546,4 +546,56 @@ keyPath.test("IUO and key paths") {
   expectEqual(kp1.hashValue, kp2.hashValue)
 }
 
+struct Subscripts<T: Hashable> {
+  var value: Int
+  subscript() -> Int {
+    get {
+      return value
+    }
+    set {
+      value = newValue
+    }
+  }
+  subscript(x: Int, y: Int) -> Int {
+    return x + y
+  }
+
+  subscript<U: Hashable>(x: T, y: U) -> (T, U) {
```

Comment:
What about a where clause here that constraints the outer parameter 'T', or relates the outer 'T' to the inner 'U'? What if 'T' or 'U' have associated types?